addons/naisjobs.md

```diff
@@ -88,65 +116,576 @@ spec:
+          name: ca-bundle-jks
+        - configMap:
+            defaultMode: 420
             name: ca-bundle-pem
-          - emptyDir:
-              medium: Memory
-            name: vault-secrets
+          name: ca-bundle-pem
+```
+
+Fetches CA bundles from `ConfigMaps` and injects it into the pod.
+
+### 2. Mount CA bundles into application container
+
+`.spec.template.spec.containers[0].volumeMounts[]`
+
+```yaml
+spec:
+  template:
+    spec:
+      containers:
+        - name: ${jobname}
+          image: ${image}
+          ...
+          volumeMounts:
+            - mountPath: /etc/ssl/certs/ca-certificates.crt
+              name: ca-bundle-pem
+              readOnly: true
+              subPath: ca-bundle.pem
+            ...
+            - mountPath: /etc/ssl/certs/java/cacerts
+              name: ca-bundle-jks
+              readOnly: true
+              subPath: ca-bundle.jks
 ```
+ 
+The files from step 1 found in `subPath` are mounted to the path specified in `path` for the container of your application.
+
+The example covers the common locations for these bundles for most systems. 
+
+### 3. (optional) Specify the location of the CA bundle
 
-If you need to read secrets from a `service user`, add the following under `spec.initContainers.volumeMounts`
+`.spec.template.spec.containers[0].env[]`
+
+```yaml
+spec:
+  template:
+    spec:
+      containers:
+        - name: ${jobname}
+          image: ${image}
+          env:
+            - name: NAV_TRUSTSTORE_PATH
+              value: /etc/ssl/certs/java/cacerts
+            - name: NAV_TRUSTSTORE_PASSWORD
+              value: changeme
 ```
-- mountPath: /var/run/secrets/nais.io/${srvUser}
-  name: vault-volume
-  subPath: vault/var/run/secrets/nais.io/${srvUser}
+
+Java applications using the [NAV baseimages] may specify environment variables to automatically inject the
+bundles/truststore into the JVM.
+
+## Vault
+
+If your job needs to access Vault for fetching secrets or uses PostgreSQL on-premises, you'll also have to
+set up Vault integration for your job.
+
+### 1. Permissions Setup
+
+To be able to retrieve secrets from Vault, you need to add permissions for the Kubernetes service account.
+
+Go to [Vault IaC] and add the following changes to your
+`terraform/teams/${teamname}/apps/${jobname}.yml` file:
+
+```yaml
+name: ${jobname}
+cluster:
+  ${cluster}:
+    serviceaccounts:
+      - default
 ```
 
-and the following under `spec.initContainers.args`
+### 2. Configuration
+
+Looking at the differences between the `Basic Example` and the `Example With Vault integration` in [Examples](#examples), 
+we spot the following:
+
+#### 2.1 Volume mount setup for Pod
+
+`.spec.template.spec.volumes[]`
+
+```yaml
+spec:
+  ttlSecondsAfterFinished: 300
+  backoffLimit: 1
+  template:
+    spec:
+      ...
+      volumes:
+        - name: vault-secrets
+          emptyDir:
+            medium: Memory
+```
+
+Sets up an empty volume mount for the pod.
```

Comment:
Lagt til `this is where the secrets will be stored in your pod, just in memory, and not written to disk.`? Eller noe i den dur?

Reply:
God ide